### PR TITLE
cmd/syncthing, lib/config: Pause/resume all devices whithout argument

### PR DESCRIPTION
**Pull request by imsodin**
### Purpose

This allows to pause/resume all devices at once via API, essentially pausing/resuming syncthing. Use case: https://forum.syncthing.net/t/syncthing-server-client-and-backup-strategy-with-rsnapshot/9256

### Testing

Just manual testing.

### Documentation

Definitely needs an update - will do, but not tonight.

### CODE REVIEW

**Comment by imsodin** [comment]
What is the state of this PR?

**Comment by AudriusButkevicius** [comment]
@st-review lgtm

**Comment by st-review** [comment]
@AudriusButkevicius: Noted! Need another LGTM or explicit merge command.

File: cmd/syncthing/gui.go [hunk 1]
```diff
@@ -1118,19 +1119,31 @@ func (s *apiService) makeDevicePauseHandler(paused bool) http.HandlerFunc {
...
+			device, err := protocol.DeviceIDFromString(deviceStr)
+			if err != nil {
+
+				http.Error(w, err.Error(), 500)
+				return
+			}
+
+			cfg, ok := s.cfg.Devices()[device]
+			if !ok {
+				http.Error(w, "not found", http.StatusNotFound)
```
Comment (calmh) [inline cmd/syncthing/gui.go]: Potential missing return here?

File: cmd/syncthing/gui.go [hunk 2]
```diff
@@ -1118,19 +1119,31 @@ func (s *apiService) makeDevicePauseHandler(paused bool) http.HandlerFunc {
...
+
+		if deviceStr == "" {
+			for _, cfg := range s.cfg.Devices() {
+				cfg.Paused = paused
+				cfgs = append(cfgs, cfg)
+			}
+		} else {
+			device, err := protocol.DeviceIDFromString(deviceStr)
+			if err != nil {
+
```
Comment (calmh) [inline cmd/syncthing/gui.go]: (spurious blank line)

**Comment by calmh** [comment]
@st-jenkins retest this please

**Comment by calmh** [comment]
@st-review lgtm

**Comment by st-review** [comment]
:ok_hand: Merged as 416811a2a99e2c04397e371774d388d6c8adb716. Thanks, @imsodin!

**Pull request closed without merging**